src/taipy/core/config/core_section.py

```diff
@@ -258,6 +273,10 @@ def _update(self, as_dict: Dict[str, Any]):
         if self.force != force:
             self.force = force
 
+        core_version = _tpl._replace_templates(as_dict.pop(self._CORE_VERSION_KEY, self._core_version))
+        if self._core_version != core_version:
+            self._core_version = core_version
+
```

Comment:
We should not make it updatable. The default value (`taipy.core.__version__`) should be populated when instantiating the Singleton and never change.
```suggestion
        core_version = _tpl._replace_templates(as_dict.pop(self._CORE_VERSION_KEY, None))
        self.check_compatibility(core_version)
```